hal/src/dmac/channel/reg.rs

```diff
@@ -243,7 +243,7 @@ macro_rules! reg_proxy {
                     // to the bit controlled by the channel.
                     self.dmac
                         .[< $reg:lower >]
-                        .modify(|r, w| unsafe { w.bits(r.bits() & ((bit as u32) << Id::U8)) });
+                        .modify(|_, w| unsafe { w.bits((bit as u32) << Id::U8) });
```

Comment:
Isn't there a just write API if you aren't going to read?

Reply:
You're right. I made a habit of using almost exclusively `modify` because that's normally what we want, but in this case since `bits` writes the whole register anyway, it doesn't really make a difference. I'm worried that would clear any other 'pending' software trigger requests though. The datasheet isn't really clear on this (ATSAMD21 19.8.1.8). thoughts?

Reply:
Actually, I realized it was just my bit manipulation logic that was wrong. [This commit](https://github.com/atsamd-rs/atsamd/pull/618/commits/bd56e9dc382782179d830d20280be30ad4326a6e) fixes that. I'm confident this can be merged now!